gatsby-node.js

```diff
@@ -99,4 +106,253 @@ async function makeSynoptic(createPage, reporter, graphql) {
       context: section
     })
   }
+}
+
+async function makeSearchIndex(reporter, graphql){
+  const lunr = require('lunr')
+  const remark = (await import('remark')).remark
+
+  const jsdom = require("jsdom")
+  const { JSDOM } = jsdom
+
+  const result_md = await graphql(`
+    query {
+      allMarkdownRemark {
+        edges {
+          node {
+            frontmatter {
+              path
+              title
+            }
+            rawMarkdownBody
+          }
+        }
+      }
+    }
+  `)
+  const result_synoptic = await graphql(`
+    query {
+      allCetei {
+        nodes {
+          prefixed
+          elements
+          parent {
+            ... on File {
+              name
+              relativeDirectory
+            }
+          }
+        }
+      }
+    }
+  `)
+
+  if (result_md.errors || result_synoptic.errors) {
+    reporter.panicOnBuild(`Error while running GraphQL query.`)
+    return
+  }
+
+
+  const searchIndex = lunr(function () {
+    this.ref('path')
+    this.field('title')
+    this.field('heading')
+    this.field('content')
+    this.metadataWhitelist = ['position']
+  
+    result_md.data.allMarkdownRemark.edges.forEach(({ node }) => {
+      const tree = remark().parse(node.rawMarkdownBody)
+      let heading = ""
+      let content = ""
+
+      for(let i = 0; i < tree.children.length; i++){
+        if(tree.children[i].type === 'heading'){
+          if(content !== ""){
+            this.add({
+              path: node.frontmatter.path,
+              title: node.frontmatter.title,
+              heading: heading,
+              content: content
+            })
+          }
+          
+          content = ""
+          heading = tree.children[i].children[0].value
+        }else{
```

Comment:
Not a big deal, we can clean up later, but it's best to add spaces between keywords and braces. `} else {` and so forth